### PR TITLE
Fix an issue with how enums are used in building Resource names.

### PR DESCRIPTION
**Pull request by AnashOommen**

### CODE REVIEW

**Review by sundquist — DISMISSED** [review]
LGTM

File: src/Util/ProtobufUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        /// </summary>
+        /// <returns>The original name, or an empty string if the name cannot be retrieved.</returns>
+        public static string GetOriginalEnumFieldName(object obj)
+        {
+            FieldInfo fi = obj.GetType().GetField(obj.ToString());
+            if (fi != null)
+            {
+                OriginalNameAttribute originalName = (OriginalNameAttribute)
+                    fi.GetCustomAttributes(typeof(OriginalNameAttribute),
+                    false).FirstOrDefault();
```
Comment (jradcliff) [inline src/Util/ProtobufUtilities.cs:38]: The indentation on lines 36-38 seems a little odd (line 37 is kinda short, and if it wraps I'd expect line 38 to be indented). Is that just how the autoformatter works?

File: src/V10/ResourceNames.cs [hunk 2]
```diff
@@ -80,7 +83,7 @@ public static string AdGroupAd(long customerId, long adGroupId, long adId)
             AssetFieldType fieldType)
         {
             return $"customers/{customerId}/adGroupAssets/{adGroupId}~{assetId}~" +
-                $"{fieldType.ToString().ToUpper()}";
+                $"{ProtobufUtilities.GetOriginalEnumFieldName(fieldType)}";
```
Comment (jradcliff) [inline src/V10/ResourceNames.cs:86]: Do you have tests for special cases like this?

File: src/Util/ProtobufUtilities.cs (hunk 1 above)
Comment (AnashOommen) [reply src/Util/ProtobufUtilities.cs]: Not sure what happened, but fixed manually.

File: src/V10/ResourceNames.cs (hunk 2 above)
Comment (AnashOommen) [reply src/V10/ResourceNames.cs:86]: Added tests.

**Pull request merged**